seacrowd/sea_datasets/visim_400/visim_400.py

```diff
@@ -0,0 +1,181 @@
+}
+"""
+
+_DATASETNAME = "visim_400"
+
+_DESCRIPTION = """\
+ViSim-400 is a Vietnamese dataset of semantic relation \
+    pairs for evaluation of models that reflect the \
+    continuum between similarity and relatedness.
+"""
+
+_HOMEPAGE = "https://www.ims.uni-stuttgart.de/forschung/ressourcen/experiment-daten/vnese-sem-datasets/"
+
+_LANGUAGES = ["vie"]
+
+_LICENSE = Licenses.CC_BY_NC_SA_2_0.value
+
+_LOCAL = False
+
+_URLS = {_DATASETNAME: "https://www.ims.uni-stuttgart.de/documents/ressourcen/experiment-daten/ViData.zip"}
+
+_SUPPORTED_TASKS = [Tasks.SEMANTIC_SIMILARITY]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class ViSim400Dataset(datasets.GeneratorBasedBuilder):
+    """
+    ViSim-400 is a Vietnamese dataset of semantic relation \
+    pairs for evaluation of models that reflect the \
+    continuum between similarity and relatedness.
+    """
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=_SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_t2t",
+            version=_SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_t2t",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+
+            features = datasets.Features(
+                {
+                    "Word1": datasets.Value("string"),
+                    "Word2": datasets.Value("string"),
+                    "POS": datasets.Value("string"),
+                    "Sim1": datasets.Value("string"),
+                    "Sim2": datasets.Value("string"),
+                    "STD": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == "seacrowd_t2t":
+            features = schemas.text2text_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        data_dir = dl_manager.download_and_extract(_URLS[_DATASETNAME])
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, f"ViData/ViSim-400/Visim-400.txt"),
```

Comment:
Unnecessary f-format.

```suggestion
                    "filepath": os.path.join(data_dir, "ViData/ViSim-400/Visim-400.txt"),
```

Reply:
Solved on https://github.com/SEACrowd/seacrowd-datahub/pull/608/commits/a9e4723635a49b82e0dd8b2fe08818007009fe9f.